packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2413,3 +2415,80 @@ const waitForSeq = async (
     };
     deltaManager.on("op", handleOp);
 });
+
+export interface ISummarizerRequestOptions {
+    cache: boolean,
+    reconnect: boolean,
+    summarizingClient: boolean,
+}
+
+/**
+ * Forms a function that will request a Summarizer.
+ * @param loaderRouter - the loader acting as an IFluidRouter
+ * @param lastSequenceNumber - the last sequence number (e.g., from DeltaManager)
+ * @param cache - use cache to retrieve summarizer
+ * @param summarizingClient - is summarizer client
+ * @param reconnect - can reconnect on connection loss
+ */
+export const formRequestSummarizerFn = (
+    loaderRouter: IFluidRouter,
+    options: ISummarizerRequestOptions,
+) => async () => {
+    const { cache, reconnect, summarizingClient } = options;
+    const request: IRequest = {
+        headers: {
+            [LoaderHeader.cache]: cache,
+            [LoaderHeader.clientDetails]: {
+                capabilities: { interactive: false },
+                type: summarizerClientType,
+            },
+            [DriverHeader.summarizingClient]: summarizingClient,
+            [LoaderHeader.reconnect]: reconnect,
+        },
+        url: "/_summarizer",
+    };
+
+    const fluidObject = await requestFluidObject<FluidObject<ISummarizer>>(loaderRouter, request);
+    const summarizer = fluidObject.ISummarizer;
+
+    if (!summarizer) {
+        return Promise.reject(new Error("Fluid object does not implement ISummarizer"));
+    }
+
+    return summarizer;
+};
+
+/**
+ * Requests a Summarizer for the purpose of performing on-demand summary.
+ * @param loader - the loader that resolves the request
+ * @param url - the URL used to resolve the container
+ * @param options - options used to resolve the container
+ * @param reconnect - can reconnect on connection loss
+ * @param summarizingClient - is summarizer client
+ */
+export async function requestOnDemandSummarizer(
+    loader: ILoader,
+    url: string,
+    options: ISummarizerRequestOptions): Promise<ISummarizer> {
+    const { cache, reconnect, summarizingClient } = options;
+    const request: IRequest = {
+        headers: {
+            [LoaderHeader.cache]: cache,
+            [LoaderHeader.clientDetails]: {
+                capabilities: { interactive: false },
+                type: summarizerClientType,
+            },
+            [DriverHeader.summarizingClient]: summarizingClient,
+            [LoaderHeader.reconnect]: reconnect,
+        },
+        url,
+    };
+
+    const resolvedContainer = await loader.resolve(request);
+    const fluidObject =
+        await requestFluidObject<FluidObject<ISummarizer>>(resolvedContainer, { url: "_summarizer" });
+    if (fluidObject.ISummarizer === undefined) {
+        throw new Error("Fluid object does not implement ISummarizer");
+    }
+    return fluidObject.ISummarizer;
```

Comment:
Given that (I think, have not finished looking through the rest of PR) you are moving toward supporting either on-demand flow or heuristic-based flow only for a given summarizer instance (i.e., not allowing usage of both from same object), I think it's worth coming up and returning here different interface that exposes on-demand summaries only here, but not for formRequestSummarizerFn() flow.

Reply:
A single summarizer can theoretically do both on-demand and heuristic-based summary. The user could ask the elected summarizer to kick off a summary on demand, and it would work. What I'm not allowing at this point is an on-demand summary using a summarizer other than the elected one, if there is an elected summarizer. (This is to avoid a race.)

Reply:
Got it. Is it obvious to developers from API shape / errors we throw what is supported / expected from them? I.e., can this be correctly figured out without need to read long documentation (that is always outdated :))?

Reply:
The error is clear, yes. The only case that may be surprising is one where the user is allowing heuristic-based summaries to happen but still wants to spin up a summarizer and run it on demand. I don't think that scenario is one we're concerned about right now, but I think the right way to handle it would be to provide a static Summarizer.summarizeOnDemand API that runs a summary using whatever summarizer is available (e.g., the elected one), creating a short-lived one if necessary.